core/include/traccc/edm/seed.hpp

```diff
@@ -46,56 +46,60 @@ struct seed {
         return *this;
     }
 
-    TRACCC_HOST_DEVICE
+    TRACCC_HOST
     std::vector<measurement> get_measurements(
         const host_spacepoint_container& spacepoints) const {
         std::vector<measurement> result;
+        result.reserve(3);
         result.push_back(spacepoints.at(spB_link).meas);
         result.push_back(spacepoints.at(spM_link).meas);
         result.push_back(spacepoints.at(spT_link).meas);
         return result;
     }
+
+    TRACCC_HOST
+    std::vector<spacepoint> get_spacepoints(
+        const host_spacepoint_container& spacepoints) const {
+        std::vector<spacepoint> result;
+        result.reserve(3);
+        result.push_back(spacepoints.at(spB_link));
+        result.push_back(spacepoints.at(spM_link));
+        result.push_back(spacepoints.at(spT_link));
+        return result;
+    }
 };
 
 inline bool operator==(const seed& lhs, const seed& rhs) {
     return (lhs.spB_link == rhs.spB_link && lhs.spM_link == rhs.spM_link &&
             lhs.spT_link == rhs.spT_link);
 }
 
-template <typename spacepoint_container_t>
-bool is_same(const seed& seed1, const spacepoint_container_t& container1,
-             const seed& seed2, const spacepoint_container_t& container2) {
+inline bool operator==(const std::vector<spacepoint>& lhs,
+                       const std::vector<spacepoint>& rhs) {
```

Comment:
If we used arrays, we could avoid the runtime for-loop from the check. It may even be possible to declare the comparison in a variadic template way where the comparison operator would not require arrays of size 3. :thinking: Though I don't know off the bat how exactly that would have to be written.

Reply:
Hm? Arrays (of arbitrary size) have a well-defined `operator==` which you could use for the comparison. Or is there something I am missing there?

Reply:
This equality operation is removed now